### DISCUSSION TITLE
Trouble Connecting gNB to the Core

### DISCUSSION THREAD

**Discussion by rolandmartinson7890** [srsran/srsRAN_Project#215] [General Help] [answered]
Hello srsRAN users

I humbly come to you after a day of trying to debug what I thought is trivial (I still believe is). It's simply just getting the core to communicate with the gNB. I had a running 5G system a week ago but decided to erase everything and install everything again. Now after following the guide on installing and connecting with open5gs, I am getting this error. I have tried different uhd versions and even different terminals.

### 

> :~/srsRAN_Project/build/apps/gnb$ sudo ./gnb -c gnb_b210_20MHz_oneplus_8t.yml
> Lower PHY in quad executor mode.
> 
> --== srsRAN gNB (commit 1afd7240f) ==--
> 
> Connecting to AMF on 127.0.1.100:38412
> No radio devices available.
> srsGNB ERROR: Unable to create radio session.
> Aborted

What I thought would solve the problem is directly copy the amf.yaml and upf.yaml and then the gnb_b210_20MHz_oneplus_8t.yml, put them in their respective directories (/etc/open5gs for amf.yaml and upf.yaml) and srsran_Project/build/apps/gnb for the gnb config files. I did this and didnt change anything and the error still persists. Ain't it supposed to be so that if we just place these already configured files in the right directories we shouldnt have to worry about configuring addresses? I did this because I thought I was messing up the addresses. 

Here is the output from the CN logs showing several attemps:

_

> **### **09/19 19:10:18.033: [sbi] INFO: [68418570-570f-41ee-98ef-b94ac1b81b7d] Subscription created until 2023-09-20T19:10:18.031456+02:00 [duration:86400,validity:86399.998167,patch:43199.999083] (../lib/sbi/nnrf-handler.c:509)
> 09/19 19:10:18.033: [sbi] INFO: [684189a8-570f-41ee-98ef-b94ac1b81b7d] Subscription created until 2023-09-20T19:10:18.031563+02:00 [duration:86400,validity:86399.998032,patch:43199.999016] (../lib/sbi/nnrf-handler.c:509)
> 09/19 19:10:18.059: [sbi] INFO: [61b1bc66-570f-41ee-8b21-b38de94f6191] (NRF-notify) NF registered (../lib/sbi/nnrf-handler.c:725)
> 09/19 19:10:18.059: [sbi] INFO: [61b1bc66-570f-41ee-8b21-b38de94f6191] (NRF-notify) NF Profile updated (../lib/sbi/nnrf-handler.c:735)
> 09/19 19:11:18.160: [amf] INFO: gNB-N2 accepted[127.0.0.1]:56383 in ng-path module (../src/amf/ngap-sctp.c:113)
> 09/19 19:11:18.160: [amf] INFO: gNB-N2 accepted[127.0.0.1] in master_sm module (../src/amf/amf-sm.c:741)
> 09/19 19:11:18.160: [amf] INFO: [Added] Number of gNBs is now 1 (../src/amf/context.c:1178)
> 09/19 19:11:18.160: [amf] INFO: gNB-N2[127.0.0.1] max_num_of_ostreams : 30 (../src/amf/amf-sm.c:780)
> 09/19 19:11:18.231: [amf] INFO: gNB-N2[127.0.0.1] connection refused!!! (../src/amf/amf-sm.c:793)
> 09/19 19:11:18.232: [amf] INFO: [Removed] Number of gNBs is now 0 (../src/amf/context.c:1205)
> 09/19 19:17:54.788: [amf] INFO: gNB-N2 accepted[127.0.0.1]:39601 in ng-path module (../src/amf/ngap-sctp.c:113)
> 09/19 19:17:54.788: [amf] INFO: gNB-N2 accepted[127.0.0.1] in master_sm module (../src/amf/amf-sm.c:741)
> 09/19 19:17:54.788: [amf] INFO: [Added] Number of gNBs is now 1 (../src/amf/context.c:1178)
> 09/19 19:17:54.789: [amf] INFO: gNB-N2[127.0.0.1] max_num_of_ostreams : 30 (../src/amf/amf-sm.c:780)
> 09/19 19:17:54.860: [amf] INFO: gNB-N2[127.0.0.1] connection refused!!! (../src/amf/amf-sm.c:793)
> 09/19 19:17:54.860: [amf] INFO: [Removed] Number of gNBs is now 0 (../src/amf/context.c:1205)
> 09/19 19:22:59.745: [amf] INFO: gNB-N2 accepted[127.0.0.1]:33415 in ng-path module (../src/amf/ngap-sctp.c:113)
> 09/19 19:22:59.745: [amf] INFO: gNB-N2 accepted[127.0.0.1] in master_sm module (../src/amf/amf-sm.c:741)
> 09/19 19:22:59.745: [amf] INFO: [Added] Number of gNBs is now 1 (../src/amf/context.c:1178)
> 09/19 19:22:59.745: [amf] INFO: gNB-N2[127.0.0.1] max_num_of_ostreams : 30 (../src/amf/amf-sm.c:780)
> 09/19 19:22:59.822: [amf] INFO: gNB-N2[127.0.0.1] connection refused!!! (../src/amf/amf-sm.c:793)
> 09/19 19:22:59.822: [amf] INFO: [Removed] Number of gNBs is now 0 (../src/amf/context.c:1205)****

_

Any help will be greatly appreciated and I hope this helps others too!

**Comment by andrepuschmann** [comment]
The issue you're having is not a core issue but the fact that you don't have a radio connected to the PC. At least the app doesn't detect one. So it's disconnecting from the core again.

**Reply by rolandmartinson7890** [reply]
Thanks for the reply @andrepuschmann . But when I do the uhd_find_usrp command, I am able to find the radio. Only srsRAN can't find the radio. 

This is happening on different PCs and after fresh reinstallations several times. Is there a certain way to install (like uhd first and then srsRAN, which is what I have been doing and I have used several uhd versions just in case one has an error.). 


Again, is it the case that I can just copy these config files, place them in the right directories and be able to run the gNB?

**Reply by rolandmartinson7890** [reply]
@andrepuschmann  another question I would ask is in which directory in Ubuntu 22.04 does srsRAN_Project expect to find the uhd files? Mayve I can move the images there.

**Comment by robertfalkenberg** [comment, marked as answer]
Whenever installing/re-installing UHD, remember to the rebuild the whole srsRAN project by deleting the `build` directory and running `cmake <your-commands>` and `make <other-commands>` again. The output of `cmake` will show you, which version was found on your system and will be incorporated in the subsequent  `make`.

**Reply by rolandmartinson7890** [reply]
Thanks @robertfalkenberg , I was able to solve the problem and I re-build the project. The problem was my method of installing UHD.
